lib/filters/filter.js

```diff
@@ -38,9 +34,17 @@ function isFilter (filter) {
   return false
 }
 
+function isBerWriter (ber) {
+  return (
+    typeof (ber) === 'object' &&
+    typeof (ber.startSequence) === 'function' &&
+    typeof (ber.endSequence) === 'function'
+  )
+}
+
 function mixin (target) {
   target.prototype.toBer = function toBer (ber) {
-    if (!ber || !(ber instanceof BerWriter)) { throw new TypeError('ber (BerWriter) required') }
+    if (!ber || !isBerWriter(ber)) { throw new TypeError('ber (BerWriter) required') }
```

Comment:
```suggestion
    if (isBerWriter(ber) === false) { throw new TypeError('ber (BerWriter) required') }
```

And make the undefined check part `isBerWriter`.

Reply:
we should check for `null` in `isBerWriter` otherwise the error `Uncaught TypeError: Cannot read property 'startSequence' of null` will be thrown.

Reply:
Yeah, that's the coerced check for `!ber`. I just want to see it in the check function instead as another inlined condition.

Reply:
Hey, I've just moved the falsiness check into the function to shorten the condition. Is this OK?